src/NuGet.Core/Microsoft.Build.NuGetSdkResolver/RestoreRunnerEx.cs

```diff
@@ -50,11 +51,21 @@ public static Task<IReadOnlyList<RestoreResultPair>> RunWithoutCommit(LibraryIde
 
                 var projectFullPath = Path.Combine(projectDirectory, $"{projectName}.proj");
 
-                // The package spec details what packages to restore
-                var packageSpec = new PackageSpec(TargetFrameworks.Select(i => new TargetFrameworkInformation
+                // Iterate through TargetFrameworks to generate Lists required for packageSpec
+                var tfi = new List<TargetFrameworkInformation>();
```

Comment:
To keep things consistent, I think you should have more verbose variable names

Reply:
Since we know the length of the list up front, might as well pass that to the constructor?